src/lib/helpers.ts

```diff
@@ -315,3 +310,40 @@ export function getMaxBlockByChainId(chainId: number, targetBlock: number) {
   // In mainnet this is always the targetBlock, but in Goerli we add 24 blocks to the targetBlock.
   return targetBlock + env.chainIdBlockOffsets[chainId];
 }
+export class PrivateKeyManager {
+  private static instance: PrivateKeyManager;
+  private keys: Record<string, string> = {};
+
+  private constructor() { }
+
+  public static getInstance(): PrivateKeyManager {
+    if (!PrivateKeyManager.instance) {
+      PrivateKeyManager.instance = new PrivateKeyManager();
+    }
+    return PrivateKeyManager.instance;
+  }
+
+  public async initialize(ovalConfigs: OvalConfigs) {
+    // Oval Config addresses are already checksummed.
+    for (const [address, config] of Object.entries(ovalConfigs)) {
+      if (config.unlockerKey) {
+        this.keys[address] = config.unlockerKey;
+      } else if (config.gckmsKeyId) {
+        const gckmsKey = await retrieveGckmsKey({
+          ...JSON.parse(env.gckmsConfig),
+          cryptoKeyId: config.gckmsKeyId,
+          ciphertextFilename: `${config.gckmsKeyId}.enc`,
+        });
+        this.keys[address] = gckmsKey;
```

Comment:
nit: can we construct the wallet and store it here?

This would avoid having plaintext keys floating around in a mapping that could accidentally get logged out. The Wallet object in ethers (while not perfect) is more purpose built for persisting a private key safely.

Reply:
Yes, good suggestion. Thank you, @mrice32.

It’s fixed!